presto-main/src/main/java/com/facebook/presto/server/security/LdapFilter.java

```diff
@@ -0,0 +1,349 @@
+
+            // ldap authentication ok, continue
+            nextFilter.doFilter(new HttpServletRequestWrapper(request)
+            {
+                @Override
+                public Principal getUserPrincipal()
+                {
+                    return principal;
+                }
+            }, servletResponse);
+        }
+        catch (AuthenticationException e) {
+            log.debug(e, "LDAP authentication failed");
+            processAuthenticationException(e, response);
+        }
+    }
+
+    private static void processAuthenticationException(AuthenticationException e, HttpServletResponse response)
+            throws IOException
+    {
+        if (e.getStatus() == UNAUTHORIZED) {
+            response.setHeader(HttpHeaders.WWW_AUTHENTICATE, "Basic realm=\"presto\"");
+        }
+        response.sendError(e.getStatus().code(), e.getMessage());
+    }
+
+    private static List<String> getCredentials(String header)
+            throws AuthenticationException
+    {
+        if (header == null) {
+            throw new AuthenticationException(UNAUTHORIZED, "Unauthorized");
+        }
+        if (!header.startsWith(BASIC_AUTHENTICATION_PREFIX)) {
+            throw new AuthenticationException(BAD_REQUEST, "Basic authentication is expected");
+        }
+        String base64EncodedCredentials = header.substring(BASIC_AUTHENTICATION_PREFIX.length());
+        String credentials = decodeCredentials(base64EncodedCredentials);
+        List<String> parts = Splitter.on(':').limit(2).splitToList(credentials);
+        if (parts.size() != 2 || parts.stream().anyMatch(String::isEmpty)) {
+            throw new AuthenticationException(BAD_REQUEST, "Invalid credentials: " + credentials);
+        }
+        return parts;
+    }
+
+    private static String decodeCredentials(String base64EncodedCredentials)
+            throws AuthenticationException
+    {
+        byte[] bytes;
+        try {
+            bytes = Base64.getDecoder().decode(base64EncodedCredentials);
+        }
+        catch (IllegalArgumentException e) {
+            throw new AuthenticationException(BAD_REQUEST, "Invalid base64 string: " + base64EncodedCredentials);
+        }
+        return new String(bytes, UTF_8);
+    }
+
+    private Principal authenticate(List<String> credentials)
+            throws AuthenticationException
+    {
+        Map<String, String> environment = createEnvironment(credentials);
+        String user = credentials.get(0);
+        InitialDirContext context = null;
+        try {
+            context = createDirContext(environment);
+            checkForGroupMembership(user, context);
+
+            log.debug("Authentication successful for user %s.", user);
+            return new LdapPrincipal(user);
+        }
+        catch (javax.naming.AuthenticationException e) {
+            String formattedAsciiMessage = format("Invalid credentials: %s", e.getMessage().replaceAll("[^\\x20-\\x7e]", ""));
```

Comment:
Yes, this was just to remove the control characters. Replaced with JAVA_ISO_CONTROL.